app/gui/view/graph-editor/src/builtin/visualization/native/text_visualization.rs

```diff
@@ -0,0 +1,469 @@
+                &init,
+                &scroll_positition,
+                &frp.set_size,
+                &text_grid.content_size,
+                f!([dom_entry_root](_, scroll_position, vis_size, content_size) {
+                    let (scroll_x, scroll_y) = *scroll_position;
+                    let top = -scroll_y * content_size.y;
+                    let bottom = top - vis_size.y;
+                    let left = scroll_x * content_size.x;
+                    let right = left +  vis_size.x;
+
+                    // Set DOM element size.
+                    dom_entry_root.set_style_or_warn("top", format!("{}px", top + PADDING_TEXT));
+                    dom_entry_root.set_style_or_warn("left", format!("{}px", -left + PADDING_TEXT));
+
+                    // Output viewport.
+                    let viewport = grid_view::Viewport {top, bottom, left, right};
+                    viewport
+                })
+            );
+            text_grid.set_viewport <+ viewport;
+        }
+
+
+        // === Style ===
+
+        let fonts_loaded = on_fonts_loaded_event(&network);
+
+        frp::extend! { network
+
+            theme_update <- all(init, font_name, font_size);
+            text_grid.set_entries_params <+  theme_update.map(
+                f!([dom_entry_root]((_, font_name, font_size)) {
+
+                    dom_entry_root.set_style_or_warn("font-family", font_name.clone());
+                    dom_entry_root.set_style_or_warn("font-size", format!("{}px", *font_size as u32));
+
+                    let parent = Some(dom_entry_root.clone_ref());
+                    grid_view_entry::Params { parent }
+                })
+            );
+
+            item_width_update <- all(init, fonts_loaded, font_name, font_size);
+            item_width <- item_width_update.map(f!([]((_, _, font_name, font_size)) {
+                let font_size = *font_size;
+                let char_width = measure_character_width(font_name, font_size);
+                CHARS_PER_CHUNK as f32 * char_width
+            })).on_change();
+            item_update <- all(init, item_width, font_size);
+            text_grid.set_entries_size <+ item_update.map(f!([]((_, item_width, item_height)) {
+                Vector2::new(*item_width, *item_height)
+            }));
+
+        }
+
+        init.emit(());
+    }
+
+    /// Set the text provider.
+    pub fn set_text_provider(&self, text_provider: T) {
+        self.model.set_text_provider(text_provider);
+        self.init_frp(self.model.text_provider.borrow().as_ref().unwrap());
+    }
+}
+
+impl<T> From<TextGrid<T>> for visualization::Instance {
+    fn from(t: TextGrid<T>) -> Self {
+        Self::new(&t, &t.frp, &t.network, Some(t.model.clipping_div.clone_ref()))
+    }
+}
+
+impl<T> display::Object for TextGrid<T> {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.root
+    }
+}
+
+
+
+// =================================
+// === Text Processing Utilities ===
```

Comment:
This title is very generic. In fact, anything can be considered an "utility". What we are doing here is measuring the glyph size.